### PR TITLE
Implement VirtualDisplayItem

### PR DESCRIPTION
**Pull request by sandtechnology**
1. Implement VirtualDisplayItem
2. Improve class structure to prevent duplicate code
3. Fix build warnings

I test it in 1.15, but it should work in 1.13+ (according to NMS code)

### CODE REVIEW

File: src/main/java/org/maxgamer/quickshop/Shop/VirtualDisplayItem.java [hunk 1]
```diff
@@ -1,3 +1,21 @@
+/*
+ * This file is a part of project QuickShop, the name is BootError.java
```
Comment (Ghost-chu) [inline src/main/java/org/maxgamer/quickshop/Shop/VirtualDisplayItem.java]: The header creared from template, if you do it by your self, you must rename the file name.

Comment (Ghost-chu) [inline src/main/java/org/maxgamer/quickshop/Shop/VirtualDisplayItem.java]: The header created from template, if you do it by your self, you must rename the file name.

**Comment by Ghost-chu** [comment]
Need add test module, test the virtual display item is works and disable when failed

**Comment by Ghost-chu** [comment]
The display won't send to player if they join the server after shop loaded / changed world, request changes.

**Comment by sandtechnology** [comment]
> 
> 
> Need add test module, test the virtual display item is works and disable when failed

But it's client side work, look like there is no way to test whether it actually works.

File: src/main/java/org/maxgamer/quickshop/Shop/VirtualDisplayItem.java [hunk 2]
```diff
@@ -63,7 +64,7 @@ public void onPacketSending(PacketEvent event) {
                 protocolManager.removePacketListener(packetListener);
                 return;
             }
-            if (!isDisplay && !isFull) {
+            if (!isDisplay && !isFull&&!Util.isLoaded(shop.getLocation())) {
```
Comment (Ghost-chu) [inline src/main/java/org/maxgamer/quickshop/Shop/VirtualDisplayItem.java]: Request changes

File: pom.xml [hunk 3]
```diff
@@ -221,7 +226,7 @@
             <artifactId>OpenInv</artifactId>
             <version>3.3.0</version>
             <scope>system</scope>
-            <systemPath>${basedir}/lib/OpenInv.jar</systemPath>
+            <systemPath>${pom.basedir}/lib/OpenInv.jar</systemPath>
```
Comment (portlek) [inline pom.xml]: `pom` should be `project`

File: pom.xml [hunk 4]
```diff
@@ -253,7 +258,7 @@
             <artifactId>clearlag</artifactId>
             <version>3.0.6</version>
             <scope>system</scope>
-            <systemPath>${basedir}/lib/Clearlag (2).jar</systemPath>
+            <systemPath>${pom.basedir}/lib/Clearlag.jar</systemPath>
```
Comment (portlek) [inline pom.xml]: `pom` should be `project`

File: pom.xml [hunk 5]
```diff
@@ -297,7 +302,7 @@
             <artifactId>Towny</artifactId>
             <version>0.95.2.8</version>
             <scope>system</scope>
-            <systemPath>${basedir}/lib/Towny-0.95.2.8.jar</systemPath>
+            <systemPath>${pom.basedir}/lib/Towny-0.95.2.8.jar</systemPath>
```
Comment (portlek) [inline pom.xml]: `pom` should be `project`

File: pom.xml [hunk 6]
```diff
@@ -322,7 +327,7 @@
             <artifactId>Residence</artifactId>
             <version>4.4.8.2</version>
             <scope>system</scope>
-            <systemPath>${basedir}/lib/Residence4.8.8.2.jar</systemPath>
+            <systemPath>${pom.basedir}/lib/Residence4.8.8.2.jar</systemPath>
```
Comment (portlek) [inline pom.xml]: `pom` should be `project`

**Comment by portlek** [comment]
Pull the master branch.

**Pull request merged**